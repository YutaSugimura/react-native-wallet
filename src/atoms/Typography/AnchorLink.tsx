import React from 'react';
import { StyleSheet, Text } from 'react-native';
import { Color, FontWeight } from '../../types/styles';
import { COLORS } from '../../common/styleguide';

const styles = StyleSheet.create({
  font: {
    textDecorationLine: 'underline',
  },
});

type Props = {
  children: React.ReactNode;
  onPress: () => void;
  fontColor?: Color;
  fontSize?: number;
  fontWeight?: FontWeight;
  underLineColor?: Color;
};

export const AnchorLink: React.FC<Props> = ({
  children,
  onPress,
  fontColor,
  fontSize,
  fontWeight,
  underLineColor,
}) => {
  const fontStyle = {
    color: fontColor ? fontColor : COLORS.FONT_LINK_BLUE,
    fontSize: fontSize ? fontSize : 14,
    fontWeight: fontWeight ? fontWeight : '400',
    lineHeight: 24,
    textDecorationColor: underLineColor ? underLineColor : COLORS.FONT_LINK_BLUE,
  };
  return (
    <Text style={[styles.font, fontStyle]} allowFontScaling={false} onPress={onPress}>
      {children}
    </Text>
  );
};
