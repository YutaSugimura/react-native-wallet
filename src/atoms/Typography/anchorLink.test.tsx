import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { AnchorLink } from './AnchorLink';

const onPress = () => {};

describe('renders correctly', () => {
  it('atoms/Typography/AnchorLink', () => {
    const tree = renderer.create(<AnchorLink onPress={onPress}>text</AnchorLink>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('atoms/Typography/AnchorLink (props)', () => {
    const tree = renderer
      .create(
        <AnchorLink
          onPress={onPress}
          fontColor="#1C1C1C"
          fontSize={20}
          fontWeight="bold"
          underLineColor="#1C1C1C"
        >
          text
        </AnchorLink>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
