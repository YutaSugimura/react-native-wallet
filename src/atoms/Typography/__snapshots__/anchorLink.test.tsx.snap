// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly atoms/Typography/AnchorLink (props) 1`] = `
<Text
  allowFontScaling={false}
  onPress={[Function]}
  style={
    Array [
      Object {
        "textDecorationLine": "underline",
      },
      Object {
        "color": "#1C1C1C",
        "fontSize": 20,
        "fontWeight": "bold",
        "lineHeight": 24,
        "textDecorationColor": "#1C1C1C",
      },
    ]
  }
>
  text
</Text>
`;

exports[`renders correctly atoms/Typography/AnchorLink 1`] = `
<Text
  allowFontScaling={false}
  onPress={[Function]}
  style={
    Array [
      Object {
        "textDecorationLine": "underline",
      },
      Object {
        "color": undefined,
        "fontSize": 14,
        "fontWeight": "400",
        "lineHeight": 24,
        "textDecorationColor": undefined,
      },
    ]
  }
>
  text
</Text>
`;
